File: 2015/ruby/day5/part2.rb
#!/usr/bin/ruby

begin
    File.open('input.txt', 'r') do |file|
        nice = 0

        file.each_line do |line|
            if not (line.match(/(\w\w).*\1/).nil? or
                    line.match(/(\w)\w\1/).nil?)
                nice += 1
            end
        end

        puts nice
    end
rescue SystemCallError
    puts 'input.txt not found.'
end

File: 2016/python/day1/part2.py
#!/usr/bin/python

class Found(Exception): pass

with open('input.txt') as file_handler:
    content = file_handler.read()

position = (0, 0)
direction = 0
directions = [(0, -1), (1, 0), (0, 1), (-1, 0)]
visited = [(0, 0)]

try:
    for instruction in content.rstrip().split(', '):
        direction = (direction + (1 if instruction[:1] == 'R' else -1)) % 4
        for step in range(0, int(instruction[1:])):
            position = tuple(map(sum, zip(position, directions[direction])))
            if position in visited: raise Found
            visited.append(position)
except Found:
    print sum(map(abs, position))
